client/scala/src/test/scala/ai/verta/client/entities/TestExperimentRun.scala

```diff
@@ -31,80 +32,172 @@ class TestExperimentRun extends FunSuite {
     f.client.close()
   }
 
-  test("getMetric should retrieve the correct logged metric") {
+  /** Test function to verify that getting should retrieve the correctly logged metadata
+   *  @param logger function to log a single piece metadata
+   *  @param multiLogger function to log multiple pieces of metadata
+   *  @param getter function to retrieve a single piece of metadata
+   *  @param allGetter function to retrieve all stored metadata
+   */
+  def testMetadata(
```

Comment:
I'd recommend adding some parenthesis in this definition. It's not clear for the unfamiliar if it will receive a function and return a value or receive a value and return a function.

Reply:
This is definitely me watching too many FP tutorials in haskell. Fixed.